### PR TITLE
[1327] Expose course level info in SubjectMapper

### PR DESCRIPTION
**Pull request by defong**
### Context
Subject of pain
A course related subjects can only fall in 1 of 3 subject level

### Changes proposed in this pull request
Make it less deadly

### Guidance to review
Nothing breaks

### CODE REVIEW

File: app/services/subject_mapper.rb [hunk 1]
```diff
@@ -215,28 +225,20 @@ def self.map_to_primary_subjects(ucas_subjects)
...
+    case subject_level
+    when :primary
       return map_to_primary_subjects(ucas_subjects)
-
-    # If the subject indicates that it's in the Further Education space,
-    # just assign Further education to it and do not associate it with any
-    # secondary subjects
-    elsif (ucas_subjects & @ucas_further_education).any?
+    when :further_education
       return ["Further education"]
```
Comment (benilovj) [inline app/services/subject_mapper.rb]:
```suggestion
       ["Further education"]
```

File: app/services/subject_mapper.rb [hunk 2]
```diff
@@ -215,28 +225,20 @@ def self.map_to_primary_subjects(ucas_subjects)
...
+    subject_level = get_subject_level(ucas_subjects);
 
-    # If the subject indicates that it's primary, do not associate it with any
-    # Secondary subjects (that happens a lot in UCAS data). Instead, mark it as primary
-    # and additionally test for specialisations (e.g. Pimary with mathematics)
-    # note a course can cover multiple specialisations, e.g. Primary with geography and Primary with history
-    elsif (ucas_subjects & @ucas_primary).any?
+    case subject_level
+    when :primary
       return map_to_primary_subjects(ucas_subjects)
```
Comment (benilovj) [inline app/services/subject_mapper.rb]:
```suggestion
       map_to_primary_subjects(ucas_subjects)
```

File: app/services/subject_mapper.rb [hunk 3]
```diff
@@ -215,28 +225,20 @@ def self.map_to_primary_subjects(ucas_subjects)
...
-    # secondary subjects
-    elsif (ucas_subjects & @ucas_further_education).any?
+    when :further_education
       return ["Further education"]
+    when :secondary
+      course_title = (course_title.strip! || course_title).downcase
 
-    # The most common case is when the course is teaching secondary subjects.
-    else
       return map_to_secondary_subjects(course_title, ucas_subjects)
```
Comment (benilovj) [inline app/services/subject_mapper.rb]:
```suggestion
       map_to_secondary_subjects(course_title, ucas_subjects)
```

**Comment by benilovj** [comment]
ah looks like the linter has picked up the same issues

**Comment by defong** [comment]
[Course - Basic detail tab Level field (backend)](https://trello.com/c/8hP7I8ic/1327-course-basic-detail-tab-level-field-backend)

**Pull request merged**